source/game.c

```diff
@@ -1919,6 +1951,25 @@ static void cards_in_hand_update_loop(bool* discarded_card, int* played_selectio
     }
 }
 
+// returns true if the top level scoring function should return too
+static bool joker_scoring_loop(int* iteration_start, Card* played_card, enum JokerEvent joker_event)
```

Comment:
```suggestion
// returns true if a joker was scored, false otherwise
static bool check_and_score_joker_for_event(int* iteration_start, Card* played_card, enum JokerEvent joker_event)
```

Rename this function, the "loop" name implies a big frame update loop like cards_in_hand_update_loop().
Also, change the comment to make this a standalone function and decouple it from the caller.